galleries/examples/user_interfaces/embedding_in_qt_sgskip.py

```diff
@@ -44,18 +44,34 @@ def __init__(self):
         self._static_ax.plot(t, np.tan(t), ".")
 
         self._dynamic_ax = dynamic_canvas.figure.subplots()
-        t = np.linspace(0, 10, 101)
         # Set up a Line2D.
-        self._line, = self._dynamic_ax.plot(t, np.sin(t + time.time()))
-        self._timer = dynamic_canvas.new_timer(50)
-        self._timer.add_callback(self._update_canvas)
-        self._timer.start()
+        self.xdata = np.linspace(0, 10, 101)
+        self._update_data()
+        self._line, = self._dynamic_ax.plot(self.xdata, self.ydata)
+        # The below two timers must be attributes of self, so that the garbage
+        # collector won't clean them after we finish with __init__...
 
-    def _update_canvas(self):
-        t = np.linspace(0, 10, 101)
+        # The data retrieval should be fast as possible (Using QRunnable should
+        # be even faster)
+        self.dataTimer = dynamic_canvas.new_timer(1)
+        self.dataTimer.add_callback(self._update_data)
+        self.dataTimer.start()
+        # Drawing at 50Hz should be fast enough for the GUI to feel smooth, and
+        # not too fast for the GUI to be overloaded with events that need to be
+        # processed while the GUI element is changed.
+        self.drawingTimer = dynamic_canvas.new_timer(20)
+        self.drawingTimer.add_callback(self._update_canvas)
+        self.drawingTimer.start()
```

Comment:
We use PEP8 names otherwise:
```suggestion
        self.data_timer = dynamic_canvas.new_timer(1)
        self.data_timer.add_callback(self._update_data)
        self.data_timer.start()
        # Drawing at 50Hz should be fast enough for the GUI to feel smooth, and
        # not too fast for the GUI to be overloaded with events that need to be
        # processed while the GUI element is changed.
        self.drawing_timer = dynamic_canvas.new_timer(20)
        self.drawing_timer.add_callback(self._update_canvas)
        self.drawing_timer.start()
```